### DISCUSSION TITLE
Linux mint arm version

### DISCUSSION THREAD

**Discussion by frankdusenbury** [linuxmint/discussions#22] [Development]
I really don't want Ubuntu... Since they killed flatpak.. I ordered a pine 64 phone I would like to run Linux mint on it and my raspberry pi.. 
Maybe I am beating a dead horse.. but it would be nice.. 
Frank

**Comment by rcalixte** [comment]
@frankdusenbury Have you considered installing [Debian/Mobian](https://wiki.debian.org/InstallingDebianOn/PINE64/PinePhone) with Cinnamon? This isn't a full one-to-one for Mint (including LMDE) but it seems like the best compromise?

**Comment by hsbasu** [comment]
Hi! You can always install Cinnamon DE on a arm based device. Binary installers are available for this. Even though [Mint's Launchpad daily build PPA](https://launchpad.net/~linuxmint-daily-build-team/+archive/ubuntu/daily-builds) does not provide the binaries, you  can use [my stable PPA](https://launchpad.net/~mamolinux/+archive/ubuntu/stable). It provides the binary installers for arm based devices.

But beware there are a few drawbacks. Cinnamon DE is not really mobile optimized. Unless you have a large screen if would be difficult to use it.

**Comment by anaximeno** [comment]
It would be nice if ARM was supported in the future so that Linux Mint could be used in M1 Macbooks or other ARM-based computers.

**Reply by rklbln** [reply]
I use an Mac with M4 an would like to have a Mint Version for Arm64. Thank you!

**Comment by duracell80** [comment]
+1 for an Armbian base.
https://www.armbian.com

Cinnamon Touch isn't needed for laptops and mini PC's, however designing themes for 4K monitors would help touchscreens (that is to say not phone screens). Gnome may be targeting phone screens and in the process simplifying their UI too much to compete with DhaliaOS. There are still a few hybrid tablet/desktop laptops.

https://dahliaos.io

What do we want? Taskbars, when do we want them? Forever! Docks, pah!!

**Comment by jcapik** [comment]
Hello guys,
so far, I have rebuilt all the packages required to successfully install the mint-meta-xfce package and the Mint additions somehow work ... of course the updates do not work as I only managed to setup a simple local repository, but at least the visual look is there ... The Firefox package from Mint is missing as it is based on a binary build from upstream and there's no binary package for arm yet, but it's always possible to install the firefox package from the ubuntu repos. There are also other minor issues like Mint Welcome showing 32-bit architecture as the detection is very probably being done like "if I'm not x86_64, then I'm i686".

I tested the installation on top of the Mantic and Jammy releases and all the dependencies were pulled by apt without problems.

I believe it would be very helpful, if the Mint team could setup an arch64 builder (running ubuntu server/mantic or armbian/jammy) and add the aarch64 repository as an experimental feature for testing purposes and of course with no guarantees ... without changing anything in the source packages and skipping packages which fail to build and of course there will be no installer needed for now. I could slowly start fixing the source packages and send patches to help with the porting. The Mint packages are very often architecture independent and work as they are.

I'm using PineA64+ with 2GB of RAM for building the packages and testing till my Pinebook Pro arrives. Virt manager also supports aarch64 and I have a second system running in emulation on x86_64, but it's unusably slow.

![linux-mint-aarch64](https://github.com/linuxmint/discussions/assets/939341/7e07c02e-14ce-4a21-a804-0067b03cb7fa)

**Reply by archisman-panigrahi** [reply]
How can I use your script? Is it hosted on GitHub?

**Comment by jcapik** [comment]
Hello guys ... I've already mentioned that in the Mint forum. I'm searching for people willing to test a script for deploying Mint on aarch64 hardware. More info is in the following forum thread:
https://forums.linuxmint.com/viewtopic.php?t=399991
Thank you

**Comment by jeroendiederen** [comment]
If there is interest, I am willing to port it to arm64. Help is needed then to get rid of Linux Mint branding. I need to have a team of developers to make new branding. It has to be very clear that such a port is unofficial and that no help can be expected from the Linux Mint team.

**Reply by marnixava** [reply]
If you could do it, that would be great.  Could an AI machine do this?  Or at least help with it?

**Reply by jeroendiederen** [reply]
I don’t need AI, I am oldschool. The porting of packages is not too difficult. My idea would be to install those Mint packages in a Debian base, preferably with debian-installer and a preseed file. I am familiar with that as I run Mint on the powerpc arch (MintPPC). There are many different arm boards, so using Armbian as base could also help. It would then later also be possible to install these packages on silica Macs if the Bananas Team has Debian support for these machines and when cinnamon works well in wayland.
I need some people who can create icons and stuff to make it look different from Linux Mint. I don’t want people to install it on their arm machines and think that it is Linux Mint and start asking support from the Linux Mint team.
This is how LMDE 6 in ppc64 looks like:
https://www.u58733p55594.web0093.zxcs-klant.nl/wp-content/uploads/2025/05/lmde6ppc.png

**Reply by tavvva** [reply]
Guys ... I'm just a bit curious. Have you tested the mintify script?

**Reply by tavvva** [reply]
Give it a try ... It should give you a smooth deployment on a supported hardware ... http://mintify.tavvva.net/

**Reply by jeroendiederen** [reply]
I never tried your script. I just built all the Mint packages for Faye (http://packages.linuxmint.com/list.php?release=faye) in an arm64 Debian Bookworm system. These packages can be installed from a repository. I have the repo running on my website.

**Comment by jeroendiederen** [comment]
I did some porting of LMDE 6 packages and it seems to work in Armbian...The beginning is there.
![armbian-mint](https://github.com/user-attachments/assets/321ec8d2-b6d6-4402-8e30-ba09aaf56a47)

**Reply by jeroendiederen** [reply]
https://github.com/linuxmint/cinnamon/issues/12937

**Comment by jeroendiederen** [comment]
Some update: I ported it to Debian as can be seen in the picture above but I experience a problem with the grouped-window-list applet. It won't load. I don't know what causes this. It might be related to the fact that I can only use an older version of cjs as I could not build the 6.0.0 version as used in LMDE 6. A simple window-list applet does work however. The advantage of a Debian base is that the whole system is lighter, very useful for the lightweight Raspberry boards. You can install LMDE 6 arm64 on top of Armbian. Armbian has a nice configuration tool available for setting up several boards.

At the moment I am trying a build in a Ubuntu Noble base, to see if that applet does work. Will keep you updated about the progress. So far I have a working cinnamon core environment, without the Mint goodies in a minimal Ubuntu cloud base. I am building the packages outside in a pbuilder sandbox and bring them in via my repository to test.

**Comment by jeroendiederen** [comment]
I now offer an installer image (ISO) of Linux Mint for arm64. The image installs the latest Linux Mint Debian Edition. This image works great in a virtual machine like Parallels Desktop. All packages which are normally in an x86-64 Cinnamon installation have been ported to arm64. This installation therefore gives a complete, very fast Linux Mint experience. Contact me for details.

**Reply by jcapik** [reply]
I'm just curious how you deal with the trademarked stuff. Because that's the main reason why nobody ever released ISO files and binary packages before ... even when there were multiple people achieving positive results with rebuilding Mint stuff.

**Comment by poyribn** [comment]
i just found out that Google-approved versions of Android will prevent sideloading in 2026.
so i am looking to replace Android with GNU/Linux, the same way i have replaced Windows with GNU/Linux.
so i hope Linux Mint will work on Android phones and tablets.

making Linux Mint work on phones seems a bit far-fetched right now, so we should start with tablets.

**Reply by jeroendiederen** [reply]
I would say try it out!

**Reply by poyribn** [reply]
i would, but Linux Mint is not supported on ARM&nbsp;devices right&nbsp;now

**Reply by viraajdd** [reply]
Yeah, Try with Tablets

**Comment by Secret-chest** [comment]
This won't fix anything. I also want LM on my telephone, but it isn't that simple. Phones are not just ARM devices; on ARM devices a lot of nonstandard components are used, and their drivers aren't in the Linux kernel itself, they use a fork. They also don't use a UEFI and this means that GRUB doesn't run there. Some more drivers are in the Android userspace and they can only run with Android. Then it is also that phone manufacturers are making it very difficult to replace the OS. Including the fact that some software updates may change the bootloader to prevent GNU/Linux from working.

The most well-known Linux phone is the PinePhone, but it is a dedicated phone specifically made to run normal Linux. There are also distros that can be installed on a few select models of Androids but support is not the best, like Droidian.org. Alternatively you could try to DIY using a board like the Raspberry Pi CM5.

**Reply by poyribn** [reply]
this is very good to know.
i noted the _Raspberry Pi Compute Module 5_

**Reply by Secret-chest** [reply]
For the CM5 you'll need to make a custom motherboard, source a screen, battery, modem, audio which is hard but not as hard as ordering ARM chips yourself. You might end up with a decently portable computer with phone function, ClockworkPi is some prior art but it can't take calls and is not a touchscreen.

**Reply by poyribn** [reply]
> For the CM5 you'll need to make a custom motherboard

i don't understand that part.
isn't the CM5 the motherboard?

> [!NOTE]
>
> in this comment thread, _CM5_ means
> _Raspberry Pi Compute Module 5_

**Reply by Secret-chest** [reply]
No, CM5 just provides the ARM chip, RAM, I/O interfaces (but not ports!), optionally WLAN/BT. It only has 200 very dense pins on the back which are connected when screwed into a suitable board that has to provide any desired ports and also power. This is some pretty basic PCB engineering, I think, but still not anyone can do it.

**Reply by Secret-chest** [reply]
There are Chinese companies who will print your PCB, that's not the problem; the design is. And I am also looking to make a similar project myself, except I also want a keyboard and I don't know what to make it out of, possibly it would be easier to use pushbuttons instead of true keyboard switches.

**Reply by poyribn** [reply]
ye, the availability for everyone to make their own electronics still has a ways to go.

i believe i will be able to make my own electronics when 500€ 3D&nbsp;printers can print a conductive material, because that would mostly automate the most difficult and tedious part of making electronics at home: precision&hyphen;aligning cables and other connections by hand.

this is the moment when an ARM64 version of Linux Mint will become relevant to me personally, for making a tablet that runs desktop software.

**Comment by jeroendiederen** [comment]
I am up to date with LMDE 7 arm64. Whenever LMDE x86-64 is released, my arm64 version will follow suit.

**Reply by archisman-panigrahi** [reply]
How did you make it?

**Comment by jeroendiederen** [comment]
You don’t unserstand what it means to produce a distribution on your 
own. I have expenses too, like website, domain name and not to mention 
the amount of time and energy which goes into producing this.

rkl schreef op 2025-10-31 18:51:

**Reply by jcapik** [reply]
Many enthusiasts understand that and even of that they do it for free regardless of expenses ;] A DONATE button is better than any kind of BUY button.

**Comment by jeroendiederen** [comment]
I might reconsider if more people were to use it. The amount of people 
using it now is less than 10.

**Reply by poyribn** [reply]
More people would use it if it was free.

Your website looks disorganized because the Buy button is hard to find. For a paid project, this makes the storefront look untrustworthy.
But if it was free, more people would believe there is no harm in trying.

I strongly suggest you make the Download and Donate buttons pretty much the first things that anyone finds on [mintppc.nl](https://mintppc.nl), not hidden in a pop&#x2011;out menu but presented right&nbsp;away &mdash; look how much easier it is to find the Download button on [linuxmint.com](https://linuxmint.com/)

**Comment by jcapik** [comment]
@sosasees well... I did it for free and people's interest isn't much better :] For it to be credible, it would have to be officially under the Mint's flag... pity :)

**Reply by poyribn** [reply]
The website can't load right now so i can't see if you reorganized it
but if you didn't, my suggestion to make the most important buttons easy to find is still important so your website looks credible and people don't get stuck searching for the actual operating&nbsp;system.
For paid software, the most important buttons are not Download&nbsp;and&nbsp;Donate but the concept is the same:
The Buy and Download buttons need to be pretty much the first things that a visitor finds on [mintppc.nl](https://mintppc.nl), immediately visible and not hidden in a (⋯)menu

**Reply by jcapik** [reply]
I'm not jeroendiederen :] We're 2 different persons.

**Comment by jeroendiederen** [comment]
LMDE 7 Gigi with the latest from zena (cinnamon 6.6) is available now.

**Comment by jeroendiederen** [comment]
Yesterday I succeeded in making a Mint-Asahi image which can be flashed onto M1/M2 Macs with an Asahi Linux installer script. The image contains a working Linux Mint system based on Debian (LMDE 7) using the Asahi kernel, m1n1 bootloader, u-boot, asahi-firmware extractor, asahi audio stack and Asahi scripts.

Included in the image is a way to set up a user account: the flashed system will boot the first time into an oem user account (autologin in LightDM), which will then autostart a small program I wrote to set up user name, password, hostname, timezone and locales. After the wizard is closed the system reboots and Linux Mint boots into a regular session where a normal user can login.

I have not been able to test this yet as I don’t have the hardware. To be continued…

**Reply by dimdom69** [reply]
Can I test this for you?

**Reply by jeroendiederen** [reply]
If you are willing to do it, sure. It is at your own risk, ok? I have never tried it yet.
Write me an email @ jjhdiederen @ zonnet dot nl
